pydocteur/github_api.py

```diff
@@ -59,3 +60,42 @@ def has_pr_number(payload) -> bool:
         return False
     else:
         return True
+
+
+def get_coauthors(pr) -> set:
+    co_authored = set()
+    compiled_regex = re.compile("(Co-authored-by:.*)")
+    for commit in pr.get_commits():
+        commit_message = commit.commit.message
+        matches = compiled_regex.findall(commit_message)
+        if matches:
+            for item in matches:
+                co_authored.add(item)
+    return co_authored
+
+
+def get_issues_to_close(body):
+    if not body or body == "":
+        return []
+    results = re.findall("(?:close[sd]?|fix|fixe[sd]|resolve[sd]?)\\s+(#\\d+)", body, flags=re.IGNORECASE)
+    return results
+
+
+def get_title_message_for_merge(pr):
+    logger.info(f"Generating title and message for merge of PR #{pr.number}")
+    co_authors = get_coauthors(pr)
+    closing_issues = get_issues_to_close(pr.body)
+
+    fixes = "Closes ".join(closing_issues)
+    coauthors = "\n".join(co_authors)
+
+    title = "Automerge of PR #{pr_number} by @{author}".format(pr_number=pr.number, author=pr.user.login)
+    message = title
+    if fixes or fixes != "":
+        message = message + "\n{fixes}".format(fixes=fixes)
+    if coauthors or coauthors != "":
+        message = message + "\n\n{coauthors}".format(coauthors=coauthors)
+    if pr.body or pr.body != "":
```

Comment:
Why doubling the `if` with an or? The empty string is false, while the non-empty string is true, so I bet this can be simplified.

Reply:
Because I'm an idiot and I tested incorrectly in the REPL